plugins/qcheck-stm/src/stm_of_ir.ml

```diff
@@ -355,18 +508,57 @@ let state_type ir =
   in
   pstr_type Nonrecursive [ td ]
 
-let cmd_type config ir =
-  let constructors = List.map (cmd_constructor config) ir.values in
+let cmd_type ir =
+  let constructors = List.map cmd_constructor ir.values in
   let td =
     type_declaration ~name:(noloc "cmd") ~params:[] ~cstrs:[]
       ~kind:(Ptype_variant constructors) ~private_:Public ~manifest:None
   in
   pstr_type Nonrecursive [ { td with ptype_attributes = [ show_attribute ] } ]
 
+let init_state config ir =
+  let pat_of_lb_arg = function
+    (* here we don't need the labels as we'll use them in the body of the function *)
+    | Gospel.Tast.Lunit -> punit
+    | Gospel.Tast.Lnone vs
+    | Gospel.Tast.Loptional vs
+    | Gospel.Tast.Lnamed vs
+    | Gospel.Tast.Lghost vs ->
+        pvar (Fmt.str "%a" Gospel.Identifier.Ident.pp vs.vs_name)
+  in
+  let bindings =
+    pexp_let Nonrecursive
+      (List.map
+         (fun (lb_arg, expr) -> value_binding ~pat:(pat_of_lb_arg lb_arg) ~expr)
+         ir.Ir.init_state.arguments)
+  in
+  let open Reserr in
+  let translate_field_desc Ir.{ model; description } =
+    let* desc = ocaml_of_term config description in
+    ok (model, desc)
+  in
+  let* fields =
+    (* XXX TODO this should be done with a fold keeping track of the identifier
```

Comment:
And this does not detect when it fails to obtain a value for one the fields, does it?